spec/requests/journal_request_spec.rb

```diff
@@ -57,4 +57,61 @@
       end
     end
   end
+
+  describe 'GET #show' do
+    let(:user) { create(:user) }
+    let(:teacher) { create(:teacher) }
+    let(:parent) { create(:parent) }
+    let!(:student) { create(:student, grade: grade) }
+    let!(:grade) { create(:grade) }
+
+    before :each do
+      sign_in current_user
+      get journal_path(id: grade.id)
+    end
+
+    # context 'with logged-in teacher' do
```

Comment:
these tests should be the same as for student's ones, the difference only in scope of returned lessons.